README.md

```diff
@@ -21,135 +21,171 @@ gem 'acme-client'
+  * [Completing an HTTP challenge](#preparing-for-http-challenge)
+  * [Completing an DNS challenge](#preparing-for-dns-challenge)
+  * [Request a challenge verification](#request-a-challenge-verification)
+  * [Downloading a certificate](#downloading-a-certificate)
+* [Extra](#extra)
+  * [Certificate revokation](#certificate-revokation)
+  * [Certificate renewal](#certificate-renewal)
 
-### Register client
+## Setting up a client
 
-In order to authenticate our client, we have to create an account for it.
+The client is initialized with a private key and the directory of your ACME provider.
+
+LetsEncrypt's `directory` is `https://acme-v01.api.letsencrypt.org/directory`
+
+`acme-ruby` expect `OpenSSL::PKey::RSA` or `OpenSSL::PKey::EC`
+
+You can generate one in Ruby using OpenSSL.
 
 ```ruby
-# We're going to need a private key.
 require 'openssl'
 private_key = OpenSSL::PKey::RSA.new(4096)
+```
+
+Or load one from a PEM file
+
+```ruby
+require 'openssl'
+OpenSSL::PKey::RSA.new(File.read('/path/to/private_key.pem'))
+```
+
+See [RSA](https://ruby.github.io/openssl/OpenSSL/PKey/RSA.html) and [EC](https://ruby.github.io/openssl/OpenSSL/PKey/EC.html) for documentation.
+
+
+```ruby
+client = Acme::Client.new(private_key: private_key, directory: 'https://acme-v01.api.letsencrypt.org/directory')
+```
+
+If your account is already registered, you can save some API call by passing you key ID directly. This will avoid unnessassary API call to retrive it from your private key.
+
+```ruby
+client = Acme::Client.new(private_key: private_key, directory: 'https://acme-v01.api.letsencrypt.org/directory', kid: 'https://example.com/acme/acct/1')
+```
+
+## Account management
+
+Account are tied to a private key. Before being allowed to create orders, the account must be registered and the ToS accepted using the private key. The account will be assigned a key ID.
+
+```ruby
+client = Acme::Client.new(private_key: private_key, directory: 'https://acme-v01.api.letsencrypt.org/directory')
+account = client.new_account(contact: 'mailto:info@example.com', terms_of_service_agreed: true)
+```
+
+## Obtaining a certificate
+### Ordering a certificate
 
-# We need an ACME server to talk to, see github.com/letsencrypt/boulder
-# WARNING: This endpoint is the production endpoint, which is rate limited and will produce valid certificates.
-# You should probably use the staging endpoint for all your experimentation:
-# endpoint = 'https://acme-staging.api.letsencrypt.org/'
-endpoint = 'https://acme-v01.api.letsencrypt.org/'
+To order a new certificate, the client must provider a list of identifiers.
 
-# Initialize the client
-require 'acme-client'
-client = Acme::Client.new(private_key: private_key, endpoint: endpoint, connection_options: { request: { open_timeout: 5, timeout: 5 } })
+The returned order will contains a list of `Authorization` that need to be completed in other to finalize the order, generally one per identifiers.
```

Comment:
"contains" to "contain"?

Since `Authorization` is singular and the context is plural, would it make sense to phrase it like "…contain a list of `Authorization` objects that need…"

"identifiers" to "identifier"?